File: SwapNumUsingPointer.java
// input: a=10, b=20 using pointer swap both number so that output: a=20, b=10

import java.util.Scanner;

public class SwapNumUsingPointer {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter the first number: ");
        int a = scanner.nextInt();
        System.out.print("Enter the second number: ");
        int b = scanner.nextInt();

        System.out.println("Before swapping: a = " + a + ", b = " + b);

        // Swapping using pointer (simulated with an array)
        int[] arr = {a, b};
        a = arr[1];
        b = arr[0];

        System.out.println("After swapping: a = " + a + ", b = " + b);
    }
}

File: SwapNumUsingPointer.c
// input: a=10, b=20 using pointer swap both number so that output: a=20, b=10

#include <stdio.h>

int main() {
    printf("Enter two numbers: ");
    int a, b;
    scanf("%d %d", &a, &b);

    int *p1 = &a, *p2 = &b;

    printf("Before swapping: a = %d, b = %d\n", a, b);

    int temp = *p1;
    *p1 = *p2;
    *p2 = temp;

    printf("After swapping: a = %d, b = %d\n", a, b);

    return 0;
}